src/error.rs

```diff
@@ -94,7 +94,8 @@ pub struct Transport {
     message: Option<String>,
     url: Option<Url>,
     source: Option<Box<dyn error::Error + Send + Sync + 'static>>,
-    response: Option<Response>,
+    #[allow(dead_code)]
+    response: Option<Response>,  // when building without tls, only tests read this.
```

Comment:
This is surprising! We should make sure we have test coverage of this code path in the non-TLS case, too.

Reply:
I realized that actually it happens even for the default, at least it does with my rustc 1.57.0-nightly (5ecc8ad84 2021-09-19)

Reply:
Aha, you are correct! Thanks for pointing that out. Now that I look at it again the reason is clear: a Response is only available on a Status error; if we got a Transport error it doesn't actually make sense to have a Response, because the error occurred before we could read a response.